### PR TITLE
Reset password entry of merge dialog before showing

### LINKED ISSUE #944: Database password is not reset when merging several databases
## Expected Behavior

When merging several databases into the currently opened database, the password field should be empty everytime another database is opened for merging.

## Current Behavior

The password entry form field contains the last used password instead of being empty.

## Possible Solution

Ensure that entered password data is not kept in memory.

## Steps to Reproduce (for bugs)

1. Open a database.
2. Merge another database and enter the corresponding database password.
3. Start merging another database.

Operating system: Mac OS X 10.12.6
CPU architecture: amd64

### PR DESCRIPTION
**Pull request by droidmonkey**
## Description

Fixes #944. Resets master key entry prior to showing the unlock dialog.

I also removed the UnlockDatabaseWidget entirely in favor of rolling everything into the base class.

## How has this been tested?

Manually and with unit tests.

## Types of changes

- ✅ Bug fix (non-breaking change which fixes an issue)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

File: src/gui/Application.cpp [hunk 1]
```diff
@@ -115,31 +115,42 @@ Application::Application(int& argc, char** argv)
...
-        // notify the other instance
-        // try several times, in case the other instance is still starting up
-        if (config()->get("SingleInstance").toBool()) {
-            QLocalSocket client;
-            for (int i = 0; i < 3; i++) {
-                client.connectToServer(socketName);
-                if (client.waitForConnected(150)) {
-                    client.abort();
-                    break;
+        case QLockFile::NoError:
```
Comment (louib) [inline src/gui/Application.cpp]: @droidmonkey I actually think not indenting the case labels is the preferred style in the project. At least that's the conclusion I came to when I was writing the `clang-format` file for the project. Either way, it'll make it easier to review if you preserve the previous style, at least in this PR.

**Comment by TheZ3ro** [comment]
> I also removed the UnlockDatabaseWidget entirely in favor of rolling everything into the base class.

I kind of agree that having a 3 line class is not ideal but I think that the code is more clear and easy to read that way (preserving the UnlockDatabaseWidget).

**Comment by droidmonkey** [comment]
It should at least follow the naming convention: DatabaseUnlockWidget

**Comment by droidmonkey** [comment]
Also, looks like I accidentally included another commit in this push, i need to amend this later today.

**Comment by TheZ3ro** [comment]
I agree for the new name `DatabaseUnlockWidget`

**Comment by droidmonkey** [comment]
@TheZ3ro why do we distinguish between "Unlock Database" and "Enter Master Key" when it is effectively the exact same thing. I think we should just make the default text "Unlock Database" and be done with it.

**Pull request closed without merging**